chains/sequential.go

```diff
@@ -0,0 +1,205 @@
+package chains
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/tmc/langchaingo/memory"
+	"github.com/tmc/langchaingo/schema"
+)
+
+// SequentialChain is a chain that runs multiple chains in sequence,
+// where the output of one chain is the input of the next.
+type SequentialChain struct {
```

Comment:
Maybe export the memory field?

Reply:
I planned to add memory support on another PR to keep this one scoped. There are some additional validations that need to occur when using memory e.g. validating there are no clashes with the memory keys and the input keys to the `SequentialChain`. 
It might be simple enough to add on this PR - can do that if you think it's worth doing.

Reply:
Yeah, didn't think about the extra validations. This PR is already kind of large so you can add it a second if you think that is best.

Reply:
I thought about this again and moved the functions to a `util` package. Given the rate of change of this project, it's probably worth bubbling up a `util` package that others can see and can take advantage of if needed.

Thanks 👍🏾